R/R/prophet.R

```diff
@@ -547,21 +548,23 @@ make_seasonality_features <- function(dates, period, series.order, prefix) {
 make_holiday_features <- function(m, dates) {
   # Strip dates to be just days, for joining on holidays
   dates <- set_date(format(dates, "%Y-%m-%d"))
-  all.holidays <- m$holidays %>% 
-    dplyr::mutate(ds=as.character(ds), holiday=as.character(holiday)) 
+  all.holidays <- m$holidays 
   if (!is.null(m$append.holidays)){
     years <- as.numeric(unique(format(dates, "%Y")))
     append.holidays.df <- make_holidays_df(years, m$append.holidays) %>%
       dplyr::mutate(ds=as.character(ds), holiday=as.character(holiday))
-    all.holidays <- all.holidays %>% dplyr::bind_rows(append.holidays.df)
+    all.holidays <- suppressWarnings(dplyr::bind_rows(all.holidays, append.holidays.df))
   }
   # Make fit.prophet and predict.prophet holidays components match
-  if (!is.null(m$train.holiday.names)){
-    row.to.drop <- which(all.holidays$holiday %in% m$train.holiday.names)
-    all.holidays <- all.holidays[row.to.drop,]
-    holidays.to.add <- data.frame(holiday=setdiff(m$train.holiday.names, 
-                                                    all.holidays$holiday)) 
-    all.holidays <- all.holidays %>% dplyr::bind_rows(holidays.to.add)
+  if (!is.null(m$append.holidays) && !is.null(m$train.holiday.names)){
+    row.to.keep <- which(all.holidays$holiday %in% m$train.holiday.names)
+    all.holidays <- all.holidays[row.to.keep,]
+    holidays.to.add <- data.frame(holiday=setdiff(m$train.holiday.names,
+                                                    all.holidays$holiday))
+    all.holidays <- suppressWarnings(dplyr::bind_rows(all.holidays, holidays.to.add))
+  }
+  if (nrow(all.holidays)==0){
+    return(NULL)
```

Comment:
I'm not a fan of returning NULL here (I prefer to have a fixed type to the function returns), but more importantly there is no check like this in the current Python function. What made this check necessary, and does it break in the current Python too?

I think the only way this would have no rows is if m$holidays is null, and m$append.holidays is not null, but the country has no holidays for the year in the training data. Is that correct?

Reply:
Yes, you are right. In the case you described, all.holidays would be NULL, and dplyr cannot operate on NULL.